exhibitions/src/main/kotlin/edu/artic/exhibitions/ExhibitionDetailFragment.kt

```diff
@@ -0,0 +1,202 @@
+import edu.artic.db.models.ArticExhibition
+import edu.artic.viewmodel.BaseViewModelFragment
+import edu.artic.viewmodel.Navigate
+import kotlinx.android.synthetic.main.fragment_exhibition_details.*
+import kotlinx.android.synthetic.main.fragment_exhibition_details.view.*
+import timber.log.Timber
+import kotlin.reflect.KClass
+
+
+class ExhibitionDetailFragment : BaseViewModelFragment<ExhibitionDetailViewModel>() {
+
+    override val viewModelClass: KClass<ExhibitionDetailViewModel>
+        get() = ExhibitionDetailViewModel::class
+    override val layoutResId: Int
+        get() = R.layout.fragment_exhibition_details
+    override val title: String
+        get() = ""
+
+    private val exhibition by lazy { arguments!!.getParcelable<ArticExhibition>(ARG_EXHIBITION) }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        postponeEnterTransition()
+        sharedElementEnterTransition =
+                TransitionInflater.from(context).inflateTransition(android.R.transition.move)
+
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        appBarLayout.addOnOffsetChangedListener { appBarLayout, verticalOffset ->
+            val progress: Double = 1 - Math.abs(verticalOffset) / appBarLayout.totalScrollRange.toDouble()
+            if (progress <= .5) {
+                val diff = .5f - progress.toFloat()
+                if (diff <= .25f) {
+                    appBarLayout.toolbarTitle.alpha = 0f
+                    appBarLayout.expandedTitle.alpha = 1 - (diff / .25f)
+                } else {
+                    appBarLayout.expandedTitle.alpha = 0f
+                    appBarLayout.toolbarTitle.alpha = (diff / .25f) - 1f
+                }
+            } else {
+                appBarLayout.expandedTitle.alpha = 0f
+                appBarLayout.expandedTitle.alpha = 1f
+            }
+
+        }
+        exhibitionImage.transitionName = exhibition.title
+    }
+
+    override fun onRegisterViewModel(viewModel: ExhibitionDetailViewModel) {
+        viewModel.setExhibitionExhibition(exhibition)
+    }
+
+    override fun setupBindings(viewModel: ExhibitionDetailViewModel) {
+        viewModel.title
+                .subscribe {
+                    expandedTitle.text = it
+                    toolbarTitle.text = it
+                }
+                .disposedBy(disposeBag)
+
+        viewModel.imageUrl
+                .subscribe {
+                    val options = RequestOptions()
+                            .dontAnimate()
+                            .dontTransform()
+                    Glide.with(this)
+//                            .asBitmap()
+                            .load(it)
+                            .apply(options)
+                            .listener(object : RequestListener<Drawable> {
+                                override fun onLoadFailed(e: GlideException?,
```

Comment:
lets move this long chain of calls into a helper method. at least convert it into an `inline` method that inlines the callbacks in a more kotlin way (less like java verbose)